### PR TITLE
[build] Make our MSBuild builder a 64bit process

### PR DESCRIPTION
**Pull request by alanmcgovern**
Xamarin.Android is 64bit only now and will only allow you to compile
Android projects if our hosted builder is also 64bit.

Fixes VSTS 724560

### CODE REVIEW

**Comment by slluis** [comment]
I think we still need the configuration property groups

File: main/src/core/MonoDevelop.Projects.Formats.MSBuild/MonoDevelop.MSBuildResolver/MonoDevelop.MSBuildResolver.csproj [hunk 1]
```diff
@@ -15,9 +15,8 @@
     <MSBuild_OSS_BinDir Condition="'$(OS)' == 'Unix' and '$(MSBuild_OSS_BinDir)' == '' and Exists('$(MSBuildToolsPath)\..\..\..\msbuild\15.0\bin\MSBuild.dll')">$(MSBuildToolsPath)\..\..\..\msbuild\15.0\bin\</MSBuild_OSS_BinDir>
     <OutputPath>..\..\..\..\build\bin\MSBuild</OutputPath>
     <DefineConstants>PUBLIC_API</DefineConstants>
+    <PlatformTarget>x64</PlatformTarget>
   </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' " />
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Projects.Formats.MSBuild/MonoDevelop.MSBuildResolver/MonoDevelop.MSBuildResolver.csproj]: These should not be removed.

File: main/src/core/MonoDevelop.Projects.Formats.MSBuild/MonoDevelop.MSBuildBuilder.csproj [hunk 2]
```diff
@@ -9,11 +9,8 @@
     <TargetFrameworkVersion>$(MDFrameworkVersion)</TargetFrameworkVersion>
     <OutputPath>..\..\..\build\bin</OutputPath>
     <StartupObject>MonoDevelop.Projects.MSBuild.MainClass</StartupObject>
+    <PlatformTarget>x64</PlatformTarget>
   </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' " />
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Projects.Formats.MSBuild/MonoDevelop.MSBuildBuilder.csproj]: These should not be removed.

File: main/src/core/MonoDevelop.Projects.Formats.MSBuild/MonoDevelop.MSBuildBuilder.csproj [hunk 3]
```diff
@@ -9,11 +9,8 @@
     <TargetFrameworkVersion>$(MDFrameworkVersion)</TargetFrameworkVersion>
     <OutputPath>..\..\..\build\bin</OutputPath>
     <StartupObject>MonoDevelop.Projects.MSBuild.MainClass</StartupObject>
+    <PlatformTarget>x64</PlatformTarget>
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Projects.Formats.MSBuild/MonoDevelop.MSBuildBuilder.csproj]: We should set this on Non-windows only.

**Comment by alanmcgovern** [comment]
i think someone else who is not me should own this work item :p

**Comment by alanmcgovern** [comment]
Hi!

This is critical issue as we cannot sanely compile our stuff anymore! Can we please get some action on this so we can get it merged asap?

**Comment by mrward** [comment]
Updated the pull request. Not managed to verify this fixes the problem. Tried master Xamarin.Android 9.1.199-11 and 9.1.199-13 and Android projects build without errors when using VS Mac 7.6 and 7.7 without this change. Is there a way to see a build failure when using the 32-bit msbuild host?

**Comment by mrward** [comment]
OK a repro is to use VS Mac 7.7, Xamarin.Android Xamarin.Android 9.1.199.11 (master) create and build an Android project. Build error occurs:

```
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(3,3): Error MSB4018: The "Javac" task failed unexpectedly.
System.DllNotFoundException: libzip.5.0.dylib
  at (wrapper managed-to-native) Xamarin.Tools.Zip.Native.zip_open(intptr,Xamarin.Tools.Zip.OpenFlags,Xamarin.Tools.Zip.ErrorCode&)
  at Xamarin.Tools.Zip.Native.zip_open (System.String path, Xamarin.Tools.Zip.OpenFlags flags, Xamarin.Tools.Zip.ErrorCode& errorp) [0x00007] in <7647a3b77ec94e7a9bd033ccf6f09f8e>:0 
  at Xamarin.Tools.Zip.ZipArchive.Open (System.String path, Xamarin.Tools.Zip.OpenFlags flags) [0x00009] in <7647a3b77ec94e7a9bd033ccf6f09f8e>:0 
  at Xamarin.Tools.Zip.ZipArchive.Open (System.String path, System.IO.FileMode mode, System.String defaultExtractionDir, System.Boolean strictConsistencyChecks, Xamarin.Tools.Zip.IPlatformOptions options) [0x0005a] in <7647a3b77ec94e7a9bd033ccf6f09f8e>:0 
  at Xamarin.Android.Tasks.ZipArchiveEx..ctor (System.String archive, System.IO.FileMode filemode) [0x0000d] in <751de2ea19194494854f3afd002d729f>:0 
  at Xamarin.Android.Tasks.Javac.Execute () [0x0003a] in <751de2ea19194494854f3afd002d729f>:0 
  at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute () [0x00023] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-06/external/bockbuild/builds/msbuild-15/src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs:573 
  at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask (Microsoft.Build.BackEnd.ITaskExecutionHost taskExecutionHost, Microsoft.Build.BackEnd.Logging.TaskLoggingContext taskLoggingContext, Microsoft.Build.BackEnd.TaskHost taskHost, Microsoft.Build.BackEnd.ItemBucket bucket, Microsoft.Build.BackEnd.TaskExecutionMode howToExecuteTask) [0x001f6] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-06/external/bockbuild/builds/msbuild-15/src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs:784  (MSB4018) 
```

**Comment by nosami** [comment]
Can verify that this PR allows VSMac to build with master Xamarin.Android   9.1.199-22

**Comment by Therzok** [comment]
Hopefully, other msbuild tasks (i.e. nuget ones) ship 64bit native libs. Looks good to me! :+1:

**Comment by Therzok** [comment]
Shouldn't matter in the long run, as macOS is depreciating 32bit

**Comment by Therzok** [comment]
@garuma maybe worth resurrecting https://github.com/mono/monodevelop/pull/1620 ?

**Comment by garuma** [comment]
@Therzok indeed although that PR seems to be handling both 32bits and 64bits cases, do we still want that or should it only be one runner that is 64bits only?

**Comment by Therzok** [comment]
> or should it only be one runner that is 64bits only?

On mac, sure. On other platforms, we need to keep that functionality, otherwise unit testing 32bit pinvokes will be broken.

**Comment by slluis** [comment]
It looks like c# binding tests are hanging.

**Comment by slluis** [comment]
@monojenkins backport release-7.7

**Pull request merged**